Fix liquid drawtype faces sometimes not rendering

Wuzzy2 opened the pull request:
Fixes #12718.

## Implementation details

The problem was, as I have suspected, that the rendering code checked `liquidtype` instead of the drawtype via `isLiquid` and `sameLiquid`. This is incorrect because liquid flowing physics are supposed to be completely independent from liquid rendering.

So the bugfix is check for the drawtype only, not liquidtype, because it's about rendering, not liquid flowing.

The `sameLiquid` function was removed because it had only one caller. It has been replaced by `sameLiquidRender`.

## How to test

Use the drawtype test nodes from DevTest (like `testnodes:liquid_3`) and surround them in glass (`testnodes:glasslike`). Viewed from every side, the liquid must look correctly.

Surround the various other liquid test nodes with glass for comparison.

Also you might test liquids in DevTest in general to make sure that other liquid features weren't accidentally broken.

SmallJoker commented on src/nodedef.h:
```diff
@@ -493,9 +493,16 @@ struct ContentFeatures
+		return (drawtype == NDT_LIQUID || drawtype == NDT_FLOWINGLIQUID);
+	}
+	bool sameLiquidRender(const ContentFeatures &f) const{
+		if (!isLiquidRender() || !f.isLiquidRender())
+			return false;
+		if (name == f.name)
+		       return true;
```
Aren't these checks superfluous? Non-liquids have the lowing ID set to `CONTENT_IGNORE`, hence the last line should suffice and is faster on top of that.

Wuzzy2 commented:
What is a "non-liquid"? The alternative liquid IDs are required for rendering, too. Have you considered nodes with a liquid drawtype but liquidtype="none"?

SmallJoker commented:
@Wuzzy2 Even though I cannot quite grasp why you would want that combination of parameters, it does not matter because the content ID assignment and `sameLiquidRender` depend on `isLiquidRender`. Hence you can compare content IDs directly because the render type "condition" is already included in its value.

Wuzzy2 commented:
So do you assume that all liquid-related nodes (be it liquids that physically flow or just nodes that only have the liquid drawtypes, without flowing) have the alternative nodes set? If yes, then the documentation needs updating as well because currently, the documentation makes it seem as that defining the "alternative" liquids is optional.

SmallJoker commented:
I do not need to assume. `liquid_alternative_flowing_id` is only set to a non-`CONTENT_IGNORE` value when `isLiquidRender() == true` (also visible in your patch). If any other behaviour should be possible, there's either an issue in the PR or my understanding.

Wuzzy2 commented:
Okay, I made a mistake in the code that sets the alternative liquid IDs. I removed the check for liquidtype, which was unreasonable. Obviously, we want the alternative liquid IDs to be set in either case: liquidtype != "none" and/or a liquid/flowingliquid drawtype. I had a brainfart, basically. :D

Now that I have changed this line again, this is now just a syntactical sugar change (you can see it in the full files diff).

Wuzzy2 commented:
I have changed the checks in `sameLiquidRender` as well. The name check is gone because string checks are slow. It was unneccessary as well.

SmallJoker reviewed (approved):
Works.

TurkeyMcMac reviewed (approved):
Looks good.
